### ISSUE TITLE
tests are not discovered

### ISSUE THREAD

**Issue by sherif414** [wallabyjs/public#3272]
### Issue description or question
i'm using vitest as my test runner, i'm having the issue of my tests not being dicoverd by wallaby but they work normaly with vitest.
the auto configuration did not work for me. wallaby just shows nothing after `Wallaby App (realtime reports) is available at: http://localhost:51245`, no tests run no errors no anything.
so i added this custom config file
```
export default () => {
  return {
    files: [
      'src/**/*.ts',
      'src/**/*.tsx',
      'src/**/*.vue',
      '!src/**/*.test.tsx',
      '!src/**/*.cy.ts',
      '!src/**/*.cy.tsx',
      '!src/**/*.test.ts',
      '!node_modules/**',
    ],
    tests: ['./src/**/*.test.ts', './src/**/*.test.tsx', '!node_modules/**'],
    env: {
      runner: 'node',
      params: {
        env: 'test=test',
      },
    },
    runMode: 'onsave',
    autoDetect: ['vitest'],
  }
}
```
now my tests are not discoverd now, here is the repo [link](https://github.com/vex-ui/primtitives)

```
{
  editorVersion: '1.75.0',
  pluginVersion: '1.0.367',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.19044',
  nodeVersion: 'v18.13.0',
  coreVersion: '1.0.1471',
  checksum: 'OTg3OWUwMjRiNmI3YTM0ZjJmMzgyZDRlNjFhMWFhN2MsMTY5NTQyNzIwMDAwMCww',
  config: {
    files: [
      { pattern: 'src/**/*.ts', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: 'src/**/*.tsx', ignore: false, trigger: true, load: true, order: 2 },
      { pattern: 'src/**/*.vue', ignore: false, trigger: true, load: true, order: 3 },
      { pattern: 'src/**/*.test.tsx', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'src/**/*.cy.ts', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'src/**/*.cy.tsx', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'src/**/*.test.ts', ignore: true, trigger: true, load: true, file: true },
      { pattern: 'node_modules/**', ignore: true, trigger: true, load: true, file: true }
    ],
    tests: [
      { pattern: 'src/**/*.test.ts', ignore: false, trigger: true, load: true, test: true, order: 4 },
      { pattern: 'src/**/*.test.tsx', ignore: false, trigger: true, load: true, test: true, order: 5 },
      { pattern: 'node_modules/**', ignore: true, trigger: true, load: true, test: true, file: false }
    ],
    env: {
      runner: 'node',
      params: { env: 'test=test', runner: '--experimental-loader=file:///c:/Users/creaz/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.367/wallabybe6a72/runners/node/hooks.mjs' },
      type: 'node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    runMode: 'onsave',
    diagnostics: {
      vitest: {
        file: {
          config: "import { fileURLToPath } from 'node:url'\r\n" +
            "import { mergeConfig } from 'vite'\r\n" +
            "import { defineConfig } from 'vitest/config'\r\n" +
            "import viteConfig from './vite.config'\r\n" +
            '\r\n' +
            'export default mergeConfig(\r\n' +
            '  viteConfig,\r\n' +
            '  defineConfig({\r\n' +
            '    test: {\r\n' +
            '      // globals: true,\r\n' +
            "      environment: 'jsdom',\r\n" +
            "      root: fileURLToPath(new URL('./', import.meta.url)),\r\n" +
            "      include: ['./src/**/*.test.{ts,tsx}'],\r\n" +
            "      exclude: ['node_modules', 'dist', '.idea', '.git', '.cache', '/playwright'],\r\n" +
            '    },\r\n' +
            '  })\r\n' +
            ')\r\n'
        },
        config: {
          config: {
            allowOnly: true,
            watch: true,
            globals: false,
            environment: 'jsdom',
            threads: true,
            clearMocks: false,
            restoreMocks: false,
            mockReset: false,
            include: [ './src/**/*.test.{ts,tsx}' ],
            exclude: [ 'node_modules', 'dist', '.idea', '.git', '.cache', '/playwright' ],
            testTimeout: 5000,
            hookTimeout: 10000,
            teardownTimeout: 10000,
            isolate: true,
            watchExclude: [ '**/node_modules/**', '**/dist/**' ],
            forceRerunTriggers: [ '**/package.json/**', '**/{vitest,vite}.config.*/**' ],
            update: false,
            reporters: [ 'default' ],
            silent: false,
            hideSkippedTests: false,
            api: { middlewareMode: true },
            ui: false,
            uiBase: '/__vitest__/',
            open: true,
            css: { include: [], modules: { classNameStrategy: 'stable' } },
            coverage: {
              provider: 'v8',
              enabled: false,
              clean: true,
              cleanOnRerun: true,
              reportsDirectory: './coverage',
              exclude: [
                'coverage/**',
                'dist/**',
                'packages/*/test?(s)/**',
                '**/*.d.ts',
                '**/virtual:*',
                '**/__x00__*',
                '**/\x00*',
                'cypress/**',
                'test?(s)/**',
                'test?(-*).?(c|m)[jt]s?(x)',
                '**/*{.,-}{test,spec}.?(c|m)[jt]s?(x)',
                '**/__tests__/**',
                '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*',
                '**/vitest.{workspace,projects}.[jt]s?(on)',
                '**/.{eslint,mocha,prettier}rc.{?(c|m)js,yml}'
              ],
              reportOnFailure: false,
              reporter: [ [ 'text', {} ], [ 'html', {} ], [ 'clover', {} ], [ 'json', {} ] ],
              extension: [
                '.js',  '.cjs',
                '.mjs', '.ts',
                '.mts', '.cts',
                '.tsx', '.jsx',
                '.vue', '.svelte'
              ],
              allowExternal: false
            },
            fakeTimers: { loopLimit: 10000, shouldClearNativeTimers: true, toFake: [ 'setTimeout', 'clearTimeout', 'setInterval', 'clearInterval', 'setImmediate', 'clearImmediate', 'Date' ] },
            maxConcurrency: 5,
            dangerouslyIgnoreUnhandledErrors: false,
            typecheck: {
              checker: 'tsc',
              include: [ '**/*.{test,spec}-d.?(c|m)[jt]s?(x)' ],
              exclude: [
                '**/node_modules/**',
                '**/dist/**',
                '**/cypress/**',
                '**/.{idea,git,cache,output,temp}/**',
                '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*'
              ]
            },
            slowTestThreshold: 300,
            root: 'C:/Users/creaz/projects/vexui/primtitives',
            config: 'C:\\Users\\creaz\\projects\\vexui\\primtitives\\vitest.config.ts',
            defines: {},
            mode: [],
            inspect: false,
            inspectBrk: false,
            singleThread: false,
            deps: {
              moduleDirectories: [ '/node_modules/' ],
              optimizer: { ssr: { enabled: true }, web: { enabled: true } },
              web: { transformAssets: true, transformCss: true, transformGlobPattern: [] }
            },
            server: { deps: { inline: [ {}, {}, {}, '@nuxt/test-utils' ], moduleDirectories: [ '/node_modules/' ], cacheDir: 'node_modules/.vitest' } },
            snapshotOptions: { snapshotFormat: {}, updateSnapshot: 'new', snapshotEnvironment: null },
            experimentalVmWorkerMemoryLimit: 2093657088,
            setupFiles: [],
            cache: { dir: 'C:/Users/creaz/projects/vexui/primtitives/node_modules/.vitest' },
            sequence: { hooks: 'parallel' },
            environmentMatchGlobs: [],
            browser: { enabled: false, headless: false, slowHijackESM: true, api: { middlewareMode: true } },
            testTransformMode: {}
          },
          projects: [
            {
              path: 'C:/Users/creaz/projects/vexui/primtitives',
              config: {
                allowOnly: true,
                watch: true,
                globals: false,
                environment: 'jsdom',
                threads: true,
                clearMocks: false,
                restoreMocks: false,
                mockReset: false,
                include: [ 'src\\**\\*.test.{ts,tsx}' ],
                exclude: [ 'node_modules', 'dist', '.idea', '.git', '.cache', 'playwright' ],
                testTimeout: 5000,
                hookTimeout: 10000,
                teardownTimeout: 10000,
                isolate: true,
                watchExclude: [ '**/node_modules/**', '**/dist/**' ],
                forceRerunTriggers: [ '**/package.json/**', '**/{vitest,vite}.config.*/**' ],
                update: false,
                reporters: [ 'default' ],
                silent: false,
                hideSkippedTests: false,
                api: { middlewareMode: true },
                ui: false,
                uiBase: '/__vitest__/',
                open: true,
                css: { include: [], modules: { classNameStrategy: 'stable' } },
                coverage: {
                  provider: 'v8',
                  enabled: false,
                  clean: true,
                  cleanOnRerun: true,
                  reportsDirectory: './coverage',
                  exclude: [
                    'coverage/**',
                    'dist/**',
                    'packages/*/test?(s)/**',
                    '**/*.d.ts',
                    '**/virtual:*',
                    '**/__x00__*',
                    '**/\x00*',
                    'cypress/**',
                    'test?(s)/**',
                    'test?(-*).?(c|m)[jt]s?(x)',
                    '**/*{.,-}{test,spec}.?(c|m)[jt]s?(x)',
                    '**/__tests__/**',
                    '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*',
                    '**/vitest.{workspace,projects}.[jt]s?(on)',
                    '**/.{eslint,mocha,prettier}rc.{?(c|m)js,yml}'
                  ],
                  reportOnFailure: false,
                  reporter: [ [ 'text', {} ], [ 'html', {} ], [ 'clover', {} ], [ 'json', {} ] ],
                  extension: [
                    '.js',  '.cjs',
                    '.mjs', '.ts',
                    '.mts', '.cts',
                    '.tsx', '.jsx',
                    '.vue', '.svelte'
                  ],
                  allowExternal: false
                },
                fakeTimers: { loopLimit: 10000, shouldClearNativeTimers: true, toFake: [ 'setTimeout', 'clearTimeout', 'setInterval', 'clearInterval', 'setImmediate', 'clearImmediate', 'Date' ] },
                maxConcurrency: 5,
                dangerouslyIgnoreUnhandledErrors: false,
                typecheck: {
                  checker: 'tsc',
                  include: [ '**/*.{test,spec}-d.?(c|m)[jt]s?(x)' ],
                  exclude: [
                    '**/node_modules/**',
                    '**/dist/**',
                    '**/cypress/**',
                    '**/.{idea,git,cache,output,temp}/**',
                    '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*'
                  ]
                },
                slowTestThreshold: 300,
                root: 'C:/Users/creaz/projects/vexui/primtitives',
                config: 'C:\\Users\\creaz\\projects\\vexui\\primtitives\\vitest.config.ts',
                defines: {},
                mode: [],
                inspect: false,
                inspectBrk: false,
                singleThread: false,
                deps: {
                  moduleDirectories: [ '/node_modules/' ],
                  optimizer: { ssr: { enabled: true }, web: { enabled: true } },
                  web: { transformAssets: true, transformCss: true, transformGlobPattern: [] }
                },
                server: { deps: { inline: [ {}, {}, {}, '@nuxt/test-utils' ], moduleDirectories: [ '/node_modules/' ], cacheDir: 'node_modules/.vitest' } },
                snapshotOptions: { snapshotFormat: {}, updateSnapshot: 'new', snapshotEnvironment: null },
                experimentalVmWorkerMemoryLimit: 2093657088,
                setupFiles: [],
                cache: { dir: 'C:/Users/creaz/projects/vexui/primtitives/node_modules/.vitest' },
                sequence: { hooks: 'parallel' },
                environmentMatchGlobs: [],
                browser: { enabled: false, headless: false, slowHijackESM: true, api: { middlewareMode: true } },
                testTransformMode: {}
              }
            }
          ],
          package: {
            version: '0.34.4',
            urls: { hooks: 'file:///c:/Users/creaz/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.367/wallabybe6a72/runners/node/hooks.mjs' },
            paths: {
              root: 'C:\\Users\\creaz\\projects\\vexui\\primtitives\\node_modules\\.pnpm\\vitest@0.34.4_@vitest+browser@0.34.4_@vitest+ui@0.34.4_jsdom@22.1.0_sass@1.67.0\\node_modules\\vitest',
              dist: 'C:\\Users\\creaz\\projects\\vexui\\primtitives\\node_modules\\.pnpm\\vitest@0.34.4_@vitest+browser@0.34.4_@vitest+ui@0.34.4_jsdom@22.1.0_sass@1.67.0\\node_modules\\vitest\\dist'
            }
          }
        }
      }
    },
    testFramework: { version: 'vitest@0.14.0', configurator: 'vitest@0.14.0', reporter: 'vitest@0.14.0', starter: 'vitest@0.14.0', autoDetected: true },
    preserveComments: false,
    extractComments: true,
    workers: { initial: 1, regular: 1, recycle: false },
    dot: true,
    captureConsoleLog: true,
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    logLimits: { inline: { depth: 5, elements: 5000 }, values: { default: { stringLength: 200 }, autoExpand: { elements: 5000, stringLength: 8192, depth: 10 } } },
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore|c8 ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    runAllTestsWhenNoAffectedTests: true,
    symlinkNodeModules: true,
    configCode: 'export default () => {\r\n' +
      '  return {\r\n' +
      '    files: [\r\n' +
      "      'src/**/*.ts',\r\n" +
      "      'src/**/*.tsx',\r\n" +
      "      'src/**/*.vue',\r\n" +
      "      '!src/**/*.test.tsx',\r\n" +
      "      '!src/**/*.cy.ts',\r\n" +
      "      '!src/**/*.cy.tsx',\r\n" +
      "      '!src/**/*.test.ts',\r\n" +
      "      '!node_modules/**',\r\n" +
      '    ],\r\n' +
      "    tests: ['./src/**/*.test.ts', './src/**/*.test.tsx', '!node_modules/**'],\r\n" +
      '    env: {\r\n' +
      "      runner: 'node',\r\n" +
      '      params: {\r\n' +
      "        env: 'test=test',\r\n" +
      '      },\r\n' +
      '    },\r\n' +
      "    runMode: 'onsave',\r\n" +
      "    autoDetect: ['vitest'],\r\n" +
      '  }\r\n' +
      '}\r\n'
  },
  packageJSON: {
    dependencies: {
      '@faker-js/faker': '^8.0.2',
      '@floating-ui/vue': '^1.0.2',
      '@heroicons/vue': '^2.0.18',
      '@tanstack/table-core': '^8.9.7',
      '@tanstack/vue-table': '^8.9.7',
      '@vueuse/core': '^10.4.1',
      chartist: '^1.3.0',
      'focus-trap': '^7.5.2',
      motion: '^10.16.2',
      vue: '3.3.4'
    },
    devDependencies: {
      '@tsconfig/node18': '^18.2.2',
      '@types/jsdom': '^21.1.3',
      '@types/node': '^18.17.17',
      '@vitejs/plugin-vue': '^4.3.4',
      '@vitejs/plugin-vue-jsx': '^3.0.2',
      '@vitest/browser': '^0.34.4',
      '@vitest/ui': '^0.34.4',
      '@vue/test-utils': '^2.4.1',
      '@vue/tsconfig': '^0.4.0',
      cypress: '^13.1.0',
      jsdom: '^22.1.0',
      'npm-run-all': '^4.1.5',
      sass: '^1.67.0',
      taze: '^0.11.2',
      typescript: '^5.2.2',
      'v-lazy-show': '^0.2.3',
      vite: '^4.4.9',
      'vite-plugin-vue-devtools': '1.0.0-rc.4',
      vitest: '^0.34.4',
      'vue-tsc': '^1.8.11'
    }
  },
  fs: { numberOfFiles: 149 },
  debug: [
    '2023-09-18T05:08:59.560Z extended-core Validating Open Source License\n',
    "2023-09-18T05:08:59.565Z extended-core Remotes:  [ 'https://github.com/vex-ui/primtitives/' ]\n",
    '2023-09-18T05:08:59.737Z config Attempting automatic configuration for vitest\n',
    '2023-09-18T05:08:59.749Z vitest/config Detected Vitest (0.34.4).\n',
    '2023-09-18T05:09:05.839Z config Finished attempting automatic configuration for vitest (6102ms)\n',
    '2023-09-18T05:09:05.844Z project Wallaby Node version: v18.13.0\n',
    '2023-09-18T05:09:05.844Z project Wallaby config: C:\\Users\\creaz\\projects\\vexui\\primtitives\\wallaby.config.js\n',
    '2023-09-18T05:09:05.861Z fs File system starting\n',
    '2023-09-18T05:09:06.331Z fs File system scan completed\n',
    '2023-09-18T05:09:06.351Z project File cache: C:\\Users\\creaz\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.367\\projects\\5fd29c02595db620\n',
    '2023-09-18T05:09:06.907Z uiService Listening port 51235\n',
    '2023-09-18T05:09:06.959Z project package.json file change detected, invalidating local cache\n',
    '2023-09-18T05:09:06.984Z workers Parallelism for initial run: 1, for regular run: 1\n',
    '2023-09-18T05:09:06.984Z workers Starting run worker instance #0\n',
    '2023-09-18T05:09:06.987Z workers Web server is listening at 53676\n',
    '2023-09-18T05:09:06.988Z project File cache requires some updates, waiting required files from IDE\n',
    '2023-09-18T05:09:07.626Z project Stopping process pool\n',
    '2023-09-18T05:09:07.630Z project Test run started; run priority: 3\n',
    '2023-09-18T05:09:07.640Z project Running all tests\n',
    '2023-09-18T05:09:07.652Z workers Starting test run, priority: 3\n',
    '2023-09-18T05:09:07.653Z nodeRunner Starting sandbox [worker #0, session #1ie83]\n',
    '2023-09-18T05:09:07.653Z nodeRunner Preparing sandbox [worker #0, session #1ie83]\n',
    '2023-09-18T05:09:07.697Z workers Started run worker instance (delayed) #0\n',
    '2023-09-18T05:09:07.699Z nodeRunner Prepared sandbox [worker #0, session #1ie83]\n',
    '2023-09-18T05:09:07.700Z workers [worker #0, session #1ie83] Running tests in sandbox\n',
    '2023-09-18T05:09:12.010Z workers Scheduling Vitest Run (1ie83): 2023-09-18T05:09:11.994Z\n',
    "2023-09-18T05:09:12.011Z workers Did not find project to provide to 'vitest' for file: C:\\Users\\creaz\\projects\\vexui\\primtitives\\src\\composables\\__tests__\\collection.test.ts\n",
    "2023-09-18T05:09:12.017Z workers Did not find project to provide to 'vitest' for file: C:\\Users\\creaz\\projects\\vexui\\primtitives\\src\\composables\\__tests__\\delayed-open.test.ts\n",
    "2023-09-18T05:09:12.019Z workers Did not find project to provide to 'vitest' for file: C:\\Users\\creaz\\projects\\vexui\\primtitives\\src\\composables\\__tests__\\selection-group.test.ts\n",
    "2023-09-18T05:09:12.019Z workers Did not find project to provide to 'vitest' for file: C:\\Users\\creaz\\projects\\vexui\\primtitives\\src\\components\\alert\\Alert.test.tsx\n",
    "2023-09-18T05:09:12.022Z workers Did not find project to provide to 'vitest' for file: C:\\Users\\creaz\\projects\\vexui\\primtitives\\src\\components\\button\\__tests__\\button.test.tsx\n",
    '2023-09-18T05:09:12.022Z workers Vitest Run Complete (1ie83): 2023-09-18T05:09:12.013Z\n',
    '2023-09-18T05:09:12.026Z workers [1ie83] Run 0 test(s), skipped 0 test(s)\n',
    '2023-09-18T05:09:12.032Z workers [1ie83] Sandbox is responsive, closing it\n',
    '2023-09-18T05:09:12.034Z project Test run finished\n',
    '2023-09-18T05:09:12.035Z project Processed console.log entries\n',
    '2023-09-18T05:09:12.036Z project Processed loading sequences\n',
    '2023-09-18T05:09:12.036Z project Processed executed tests\n',
    '2023-09-18T05:09:12.036Z project Processed code coverage\n',
    '2023-09-18T05:09:12.044Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by smcenlly** [comment]
Thanks for reporting the problem. We believe we have found and fixed your problem. Could you please update to Wallaby core `v1.0.1472`, it should now work for you.

**Comment by sherif414** [comment]
i can confirm its working now. thanks alot

**Comment by smcenlly** [comment]
We noticed another problem with your project that we just finished investigating. For some reason, Wallaby was very slow to start when using automatic configuration without your `wallaby.config.js` file.

We discovered that your `vitest.config.ts` patterns are no longer optimised for fast file matching.

By default, `vitest` exclude patterns are:

```javascript
[
  "**/node_modules/**",
  "**/dist/**",
  "**/cypress/**",
  "**/.{idea,git,cache,output,temp}/**",
  "**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*",
]
```

Your configuration overrides these patterns, and sets them to:

```javascript
[
  "node_modules",
  "dist",
  ".idea",
  ".git",
  ".cache",
  "/playwright",
]
```

The trailing `/**` is missing from your exclude patterns, which causes additional file system process. We recommend adding the trailing `/**`, it should speed up Wallaby and remove the need for overriding the `files` and `tests` configuration (allowing you to have a single file/tests configuration in your `vitest.config.ts` file).
